src/include/job.h

```diff
@@ -853,14 +855,15 @@ typedef struct	infoent {
 #define IM_RESTART		16
 #define IM_DELETE_JOB		17
 #define IM_REQUEUE		18
-#define	IM_DELETE_JOB_REPLY	19
+/* skip 19 to avoid confilict with IS_DISCARD_JOB */
```

Comment:
Why is this change needed? One is inter-mom message and the other is inter-server message. How can those two conflict?

Reply:
@arungrover Thank you for reviewing and providing the comments.

> @riyazhakki Can you please explain a little more about your change and about what happens in this race condition.

During execution of the epilogue hook if the user hits the 'rerun -W force job_id' then the server sends IS_DISCARD_JOB to mom (mom kills the job and starts the exec_jobend hook in the background) and force queue at #L469, in this case it get scheduled to the same exec_vnode where the previous version was running. While the mom starts running hook in the background at the same time it receives the request to rerun the next version of same job, as mom is not blocked for running exec_jobend (background) hook.
When this job version lands on mom, req_commit() will change its state to 41 and start_exec function will abort it, as it sees its previous version is not yet cleaned up. Later, after execution of execjob_end hook (of old job version) mom will purge the job structure and on the server it remains in 41 state and similarly it happens with qrun also.
 

> Why is this change needed? One is inter-mom message and the other is inter-server message. How can those two conflict?

Sister node can receive both messages IM_DELETE_JOB_REPLY and IS_DISCARD_JOB (in case of node fail requeue), and replying to these request is done in a function reply_hook_bg(), therefore the above change is required.